### PR TITLE
Destination Redshift: Json transformer for graceful handling of large records

### PR DESCRIPTION
**Pull request by gisripa**
Summary
=======
Transformer implementation, In-place nulling and avoiding double walking the tree to get totalBytes. 
This also picks up previous unpublished CDK changed; Kotlin conversion & State messages log downgraded to DEBUG level. 

- [x] Wiring StagingConsumerFactory, JdbcBufferedConsumerFactory
- [x] Integration test to capture large record simulation

### CODE REVIEW

**Comment by gisripa** [comment]
[Staging upload Run](https://cloud.airbyte.com/workspaces/b61bc266-ef3c-460c-af2f-f70da4a2993c/connections/e1e1f841-ca63-4524-9c87-9da81f928a79/job-history#9440831::0), [Standard inserts run](https://cloud.airbyte.com/workspaces/b61bc266-ef3c-460c-af2f-f70da4a2993c/connections/4e5476f3-79d6-4541-810a-4666f4e156c0/job-history#9465250::0)

File: airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftSuperLimitationTransformer.java [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Predicate;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+
+@Slf4j
```
[the remark this answers is not in the document]
Comment (jbfbell) [reply airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftSuperLimitationTransformer.java:40]: Looking for something like a "_Here be dragons..._" comment

**Review by jbfbell — APPROVED** [review]

File: airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftSuperLimitationTransformer.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    final List<AirbyteRecordMessageMetaChange> changes = new ArrayList<>();
+
+    // This was intentionally done using Iterative DFS to avoid stack overflow for large records.
+    // This will ensure we are allocating on heap and not on stack.
+    stack.push(ImmutablePair.of("$", rootNode));
+    while (!stack.isEmpty()) {
+      final ImmutablePair<String, JsonNode> jsonPathNodePair = stack.pop();
+      final JsonNode currentNode = jsonPathNodePair.right;
+      if (currentNode.isObject()) {
+        originalBytes += getByteSize("{}");
```
[the remark this answers is not in the document]
Comment (gisripa) [reply airbyte-integrations/connectors/destination-redshift/src/main/java/io/airbyte/integrations/destination/redshift/typing_deduping/RedshiftSuperLimitationTransformer.java]: done.

**Pull request merged**